rockchip: rk3328: add support for FriendlyARM NanoPi NEO3

mj22226 opened the pull request:
```
NanoPi NEO3 Specification:

 Soc:      RockChip RK3328
 RAM:      1GB/2GB DDR4
 LAN:      10/100/1000M Ethernet with unique MAC
 USB Host: 1x USB3.0 Type A and 2x USB2.0 pin header
 MicroSD:  x 1 for system boot and storage
 LED:      Power LED x 1, System LED x 1
 Key:      User Button x 1
 Fan:      2 Pin JST ZH 1.5mm Connector for 5V Fan
 GPIO:     26 pin-header, include I2C, UART, SPI, I2S, GPIO
 Power:    5V/1A, via Type-C or GPIO
```
The NanoPi NEO3 works with the NanoPi R2S firmware image, the issue comes up that the R2S has its LAN on eth1, so setup has to be made via TTL serial port or with a RTL8152 USB Ethernet Adapter acting as eth1.

1715173329 commented:
Just remove the extra definitions in Neo3 DTS.
See https://elinux.org/Device_Tree_Source_Undocumented#Deleting_nodes_and_properties

blocktrron commented on target/linux/rockchip/patches-5.4/105-rockchip-rk3328-add-support-for-FriendlyARM-NanoPi-N.patch:
```diff
@@ -0,0 +1,75 @@
+Date: Sat, 26 Dec 2020 05:08:15 -0500
+Subject: [PATCH] rockchip:rk3328: add support for FriendlyARM NanoPi NEO3
+
+This patch adds support for FriendlyARM NanoPi NEO3
```
Please send this upstream.

mj22226 replied:
@blocktrron I sent the patch uptream [https://lore.kernel.org/linux-arm-kernel/20201228152836.02795e09.mj8263788@gmail.com/](url)

blocktrron replied:
linux-rockchip / linux-arm tends to ignore your patches if they are not sent to the maintainers themselves. So depending on whether this is still the case or not, you might need to resent the patch to a broader audience.

blocktrron commented on target/linux/rockchip/armv8/base-files/etc/board.d/01_leds:
```diff
@@ -13,6 +13,8 @@ friendlyarm,nanopi-r2s)
 	ucidef_set_led_netdev "wan" "WAN" "$boardname:green:wan" "eth0"
 	ucidef_set_led_netdev "lan" "LAN" "$boardname:green:lan" "eth1"
 	;;
+friendlyarm,nanopi-neo3)
```
Alphabetical sorting

mj22226 committed:
rockchip: add support for FriendlyARM NanoPi NEO3

 Specification:

 Soc:      RockChip RK3328
 RAM:      1GB/2GB DDR4
 LAN:      10/100/1000M Ethernet with unique MAC
 USB Host: 1x USB3.0 Type A and 2x USB2.0 pin header
 MicroSD:  x 1 for system boot and storage
 LED:      Power LED x 1, System LED x 1
 Key:      User Button x 1
 Fan:      2 Pin JST ZH 1.5mm Connector for 5V Fan
 GPIO:     26 pin-header, include I2C, UART, SPI, I2S, GPIO
 Power:    5V/1A, via Type-C or GPIO

 Firmware Installation
 ======================
 gzip -d xxx.img.gz, then dd the .img to SD
 ======================

Signed-off-by: Marty Jones <mj8263788@gmail.com>

adschm commented on target/linux/rockchip/armv8/base-files/etc/board.d/01_leds:
```diff
@@ -9,6 +9,9 @@ boardname="${board##*,}"
 
 case $board in
+friendlyarm,nanopi-neo3)
+	ucidef_set_led_netdev "lan" "STAT" "$boardname:green:sys" "eth0"
```
So, is it "lan", "stat" or "sys"?

adschm replied:
Since this appears to have lan _and_ wan LEDs, one should use these appropriately.

1715173329 replied:
> So, is it "lan", "stat" or "sys"?

Howerver, there's only a LED on NEO3. In accordance with practice, maybe
it's more fair to set it as "sys."

adschm replied:
And why do you add three LEDs in your kernel patch then?

1715173329 replied:
> And why do you add three LEDs in your kernel patch then?

This patch is outdated and just needs be updated by author.
Also, he had send it to kernel, just ping him and see what's new.

adschm replied:
Well, this is your submission, so it needs to be updated by you?

1715173329 replied:
> Well, this is your submission, so it needs to be updated by you?

Nope, this was actually created by @mj22226.

@mj22226 Could you please have a look? or is it still in wip?

adschm replied:
Well, somebody will have to make it proper, it's not so important who that will be ...

1715173329 replied:
As I don't have such a device, I'm sorry for that I cannot really take a look on it.

adschm replied:
@1715173329 Sorry, I was persistently under the impression that you were the author. Sorry for my tone and insistence and thanks for providing the documentation.

1715173329 replied:
@adschm Never mind, please.
Making codes better is what we always do, so yes, I just saw a conscientious and responsible
developer, but nothing else.

Also, I'm glad if I could do anything for it :-)

adschm commented on target/linux/rockchip/image/armv8.mk:
```diff
@@ -5,6 +5,15 @@
+  DEVICE_MODEL := NanoPi NEO3
+  SOC := rk3328
+  UBOOT_DEVICE_NAME := nanopi-r2s-rk3328
+  IMAGE/sysupgrade.img.gz := boot-common | boot-script nanopi-r2s | pine64-img | gzip | append-metadata
```
Please wrap line after 74 chars.

adschm commented on target/linux/rockchip/patches-5.4/105-rockchip-rk3328-add-support-for-FriendlyARM-NanoPi-N.patch:
```diff
@@ -0,0 +1,75 @@
+--- /dev/null
++++ b/arch/arm64/boot/dts/rockchip/rk3328-nanopi-neo3.dts
+@@ -0,0 +1,35 @@
++// SPDX-License-Identifier: (GPL-2.0+ OR MIT)
```
GPL-2.0-or-later OR MIT

adschm commented on target/linux/rockchip/patches-5.4/105-rockchip-rk3328-add-support-for-FriendlyARM-NanoPi-N.patch:
```diff
@@ -0,0 +1,75 @@
++ */
++
++/dts-v1/;
++#include "rk3328-nanopi-r2s.dts"
```
That's bad practice. You should not derive one DTS from another.

adschm commented on target/linux/rockchip/patches-5.4/105-rockchip-rk3328-add-support-for-FriendlyARM-NanoPi-N.patch:
```diff
@@ -0,0 +1,75 @@
++		lan_led: led-0 {
++			gpios = <&gpio2 RK_PB7 GPIO_ACTIVE_HIGH>;
++			label = "nanopi-neo3:green:lan";
++			status = "disabled";
```
Why disabled?

1715173329 replied:
> Why disabled?

This is not exist, so it should be droped.

adschm replied:
Well, then remove the node instead of disabling it.

gytisgreitai commented:
@mj22226 are you planning on finishing this?

adschm commented:
This is obviously abandoned.

Please ask for reopening when you want to continue the work.

robnelsonch commented:
Could we please reopen this? Thank you.
